frontend/src/services/auth.tsx

```diff
@@ -14,6 +14,8 @@ class AuthService {
   private tokenSubject$ = new BehaviorSubject<string | null>(
     localStorageService.getToken()
   );
+  private registeredUser$ = new BehaviorSubject<string | null>(null);
+  private wantsToLogin$ = new BehaviorSubject<boolean>(true);
```

Comment:
Using boolean can be confusing in case like that. Good thing that it's usage is hidden behind `chooseLogin` and `chooseRegister` methods, so it is not visible outside of component. However it would be better if you'd create enum e.g. 
```
enum AuthStage {
    Login,
    Register
}
```
Then the subject should look like this:
```
 private authStage$ = new BehaviorSubject<AuthStage>(AuthStage.Login);
```

Reply:
Please check if my solution is correct.